apps/examples/testcase/ta_tc/ble_manager/utc/utc_blemanager_main.c

```diff
@@ -0,0 +1,1128 @@
+		scanned_device->addr.mac[2],
+		scanned_device->addr.mac[3],
+		scanned_device->addr.mac[4],
+		scanned_device->addr.mac[5]
+	);
+	return;
+}
+
+static void ble_device_disconnected_cb(ble_client_ctx *ctx)
+{
+	printf("client disconnected callback received\n");
+	g_client_connected = BEVT_DISCONNECTED;
+	UTC_FUNC_SIGNAL;
+	return;
+}
+
+static void ble_device_connected_cb(ble_client_ctx *ctx, ble_device_connected *dev)
+{
+	printf("client connected callback received\n");
+	g_client_connected = BEVT_CONNECTED;
+	UTC_FUNC_SIGNAL;
+	return;
+}
+
+static void ble_operation_notification_cb(ble_client_ctx *ctx, ble_attr_handle attr_handle, ble_data *read_result)
+{
+	printf("notification callback received\n");
+	g_client_noti = BEVT_NOTI;
+	UTC_FUNC_SIGNAL;
+	return;
+}
+
+static void ble_server_connected_cb(ble_conn_handle con_handle, ble_server_connection_type_e conn_type, uint8_t mac[BLE_BD_ADDR_MAX_LEN])
+{
+	printf("server connect callback received");
+	return;
+}
+
+static ble_server_gatt_t gatt_profile[] = {
+	{
+		.type = BLE_SERVER_GATT_SERVICE,
+		.uuid = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x01, 0x01, 0x01, 0x01},
+		.uuid_length = 16,
+		.attr_handle = 0x006a,
+	},
+
+	{
+		.type = BLE_SERVER_GATT_CHARACT,
+		.uuid = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x01, 0x01, 0x01, 0x02},
+		.uuid_length = 16,
+		.property = BLE_ATTR_PROP_RWN | BLE_ATTR_PROP_WRITE_NO_RSP,
+		.permission = BLE_ATTR_PERM_R_PERMIT | BLE_ATTR_PERM_W_PERMIT,
+		.attr_handle = 0x006b,
+		.arg = "char_a_1"
+	},
+
+	{
+		.type = BLE_SERVER_GATT_DESC,
+		.uuid = {0x02, 0x29},
+		.uuid_length = 2,
+		.permission = BLE_ATTR_PERM_R_PERMIT | BLE_ATTR_PERM_W_PERMIT,
+		.attr_handle = 0x006c,
+		.arg = "desc_b_1",
+	},
+};
+
+static uint8_t g_adv_raw[] = {
+	0x02, 0x01, 0x05, 0x03, 0x19, 0x80, 0x01, 0x05, 0x03, 0x12, 0x18, 0x0f, 0x18
+};
+static uint8_t g_adv_resp[] = {
+	0x11, 0x09, 'T', 'I', 'Z', 'E', 'N', 'R', 'T', ' ', 'T', 'E', 'S', 'T', '(', '0', '2', ')',
+};
+
+static ble_scan_callback_list scan_config = {
+	ble_scan_state_changed_cb,
+	ble_device_scanned_cb,
+};
+
+static ble_client_callback_list client_config = {
+	ble_device_disconnected_cb,
+	ble_device_connected_cb,
+	ble_operation_notification_cb,
+};
+
+static ble_server_init_config server_config = {
+	ble_server_connected_cb,
+	true,
+	gatt_profile,
+	sizeof(gatt_profile) / sizeof(ble_server_gatt_t)
+};
+
+static void set_scan_filter(ble_scan_filter *filter, uint8_t *raw_data, uint8_t len, bool whitelist_enable, uint32_t scan_duration)
+{
+	memset(filter, 0, sizeof(ble_scan_filter));
+	if (raw_data != NULL && len > 0) {
+		memcpy(filter->raw_data, raw_data, len);
+		filter->raw_data_length = len;
+	}
+
+	filter->scan_duration = scan_duration;
+	filter->whitelist_enable = whitelist_enable;
+}
+
+static void utc_blemanager_init_p(void)
+{
+	TC_ASSERT_EQ("ble_manager_init", ble_manager_init(NULL), BLE_MANAGER_SUCCESS);//not server role
```

Comment:
Could you please elaborate in comments what is not server role/server role and why do we need this ?

Reply:
BLE manager can be initiated in two configurations client & server.
When passed NULL it takes the server role and you can use server specific APIs like advertising.